### ISSUE TITLE
query result error

### ISSUE THREAD

**Issue by wanpengfei-git** [StarRocks/starrocks#3779]
### Steps to reproduce the behavior (Required)
1. DROP DATABASE IF EXISTS g1;
CREATE DATABASE g1;
USE g1;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE t0 (c_0_0 TINYINT NULL ,c_0_1 CHAR(1) NULL ,c_0_2 DECIMAL(18, 3)  NOT NULL ,c_0_3 FLOAT MAX NULL ,c_0_4 DOUBLE MIN NULL ,c_0_5 PERCENTILE PERCENTILE_UNION NULL ,c_0_6 DATE REPLACE_IF_NOT_NULL NULL ,c_0_7 DATETIME REPLACE NULL ,c_0_8 DOUBLE MIN NOT NULL ,c_0_9 DATETIME MIN NULL ,c_0_10 VARCHAR(21) MAX NULL ,c_0_11 DECIMAL(18, 1)  REPLACE_IF_NOT_NULL NOT NULL ) AGGREGATE KEY (c_0_0,c_0_1,c_0_2) DISTRIBUTED BY HASH (c_0_2,c_0_1) properties("replication_num"="3") ;
CREATE TABLE IF NOT EXISTS t1 (c_1_0 BOOLEAN NOT NULL ,c_1_1 VARCHAR(21) NOT NULL ,c_1_2 DECIMAL(4, 3)  NOT NULL ,c_1_3 DECIMAL(9, 4)  NULL ,c_1_4 LARGEINT NULL ,c_1_5 DECIMAL(15, 7)  NULL ,c_1_6 BOOLEAN NULL ,c_1_7 DECIMAL(5, 2)  MAX NULL ,c_1_8 SMALLINT REPLACE NULL ,c_1_9 CHAR(1) REPLACE NOT NULL ,c_1_10 DECIMAL(26, 7)  REPLACE NULL ,c_1_11 BITMAP BITMAP_UNION NOT NULL ,c_1_12 DATETIME MAX NULL ,c_1_13 PERCENTILE PERCENTILE_UNION NOT NULL ,c_1_14 DATETIME MIN NOT NULL ) AGGREGATE KEY (c_1_0,c_1_1,c_1_2,c_1_3,c_1_4,c_1_5,c_1_6) DISTRIBUTED BY HASH (c_1_5,c_1_0,c_1_1) properties("replication_num"="3") ;
CREATE TABLE IF NOT EXISTS t2 (c_2_0 VARCHAR(1) NOT NULL ,c_2_1 CHAR(31) MAX NOT NULL ,c_2_2 BITMAP BITMAP_UNION NULL ,c_2_3 PERCENTILE PERCENTILE_UNION NOT NULL ,c_2_4 CHAR(11) MAX NOT NULL ,c_2_5 VARCHAR(11) REPLACE_IF_NOT_NULL NOT NULL ,c_2_6 BITMAP BITMAP_UNION NOT NULL ,c_2_7 DATETIME MAX NULL ,c_2_8 CHAR(11) MAX NOT NULL ,c_2_9 CHAR(11) REPLACE NOT NULL ) AGGREGATE KEY (c_2_0) DISTRIBUTED BY HASH (c_2_0) properties("replication_num"="3") ;
CREATE TABLE IF NOT EXISTS t3 (c_3_0 BOOLEAN NULL ,c_3_1 BOOLEAN NOT NULL ,c_3_2 BOOLEAN NULL ,c_3_3 BIGINT NOT NULL ,c_3_4 DOUBLE NULL ) UNIQUE KEY (c_3_0,c_3_1,c_3_2,c_3_3) DISTRIBUTED BY HASH (c_3_3) properties("replication_num"="3") ;
INSERT INTO t0 VALUES (6, "", 250013521776.522, 0.799856266731065, -1.7976931348623157E308, percentile_hash("-1358676598"), "1970-01-01", "1970-01-01 08:00:00", 0.7279249833148909, NULL, "", 6059172670050370.1), (1, "", 644274782386.372, 0.7279249833148909, 0.20998016702152345, percentile_hash("-1140818424"), NULL, "1970-01-01 08:00:00", -1.140818424E9, "1970-01-01 08:00:00", "", 6161966487510288.8), (3, NULL, 401163931715.156, 0.6160339673022679, 0.9799287594925228, percentile_hash("-410627537"), "1970-01-01", "1970-01-01 08:00:00", 0.4495190388004586, "1970-01-01 08:00:00", "xvfl", NULL);
INSERT INTO t1 VALUES (false, "zzrWbA", 0.007, 9.8763, -410627537, NULL, false, 6.55, NULL, "", NULL, bitmap_hash("ty8wy"), "1970-01-01 08:00:00", percentile_hash("-1151442401"), "1970-01-01 08:00:00"), (true, "vPo", 0.006, 5.3094, -1358676598, 8.7256019, false, 7.01, 0, "", 275058634704.3083177, bitmap_hash("b"), NULL, percentile_hash("-1151442401"), "1970-01-01 08:00:00");
INSERT INTO t3 VALUES (false, false, NULL, -1358676598, 0.9098722713217339), (true, true, true, 931274810, -1.358676598E9);
INSERT INTO t0 VALUES (NULL, "", 610139433592.297, 0.07964081652859778, 0.14438117594919153, percentile_hash("-410627537"), "1970-01-01", "1970-01-01 08:00:00", 0.8574513494701398, "1970-01-01 08:00:00", "QPL", NULL), (NULL, "", 15793050972.017, 0.15695980840433132, 0.7313188597283709, percentile_hash("106912963"), "1970-01-01", "1970-01-01 08:00:00", -1.151442401E9, "1970-01-01 08:00:00", "", 9261841377977160.2);
INSERT INTO t1 VALUES (false, "MKO", 0.005, 4.7698, 1766800346, NULL, false, 8.61, 2, "", 877002986620.4202058, bitmap_hash("Y"), "1970-01-01 08:00:00", percentile_hash("1229531886"), "1970-01-01 08:00:00"), (false, "", 0.007, 7.5445, 1501589653, 1.8763711, false, 4.11, 10, "", 816757448985.8643167, bitmap_hash("ZX"), NULL, percentile_hash("-1151442401"), "1970-01-01 08:00:00");
INSERT INTO t0 VALUES (0, "", 829756864012.930, 0.5800027235762893, 0.799856266731065, percentile_hash("-1151442401"), NULL, NULL, 1.766800346E9, "1970-01-01 08:00:00", "N", 914015685147036.7);
INSERT INTO t1 VALUES (false, "71mvzxWY", 0.006, 3.5782, -2120669330, 3.5040121, true, NULL, 5, "", 812214887758.9759103, bitmap_hash(""), "1970-01-01 08:00:00", percentile_hash("-1151442401"), "1970-01-01 08:00:00"), (true, "V", 0.008, 0.3370, -846957675, 0.7798002, NULL, NULL, NULL, "", 931842064022.9900426, bitmap_hash(""), "1970-01-01 08:00:00", percentile_hash("1662323466"), "1970-01-01 08:00:00");
INSERT INTO t1 VALUES (true, "I", 0.003, 7.1768, 1501589653, NULL, true, 9.17, 1, "", 110382184289.3349505, bitmap_hash("L"), NULL, percentile_hash("-1151442401"), "1970-01-01 08:00:00");
DELETE FROM t1 WHERE c_1_2=0.004 AND c_1_0=true AND c_1_5=6.2476707;
INSERT INTO t0 VALUES (NULL, "", 959529333735.426, 0.3731812048561487, 0.283279680902655, percentile_hash("650339947"), "1970-01-01", "1970-01-01 08:00:00", 0.18407982299507275, "1970-01-01 08:00:00", "6o", 481867677210892.4), (1, "", 398203356112.521, NULL, NULL, percentile_hash("-2043233926"), "1970-01-01", "1970-01-01 08:00:00", 0.439457174104408, "1970-01-01 08:00:00", "SZ", 7467086401094956.1);
INSERT INTO t2 VALUES ("", "HZ", bitmap_hash("5l9JqH"), percentile_hash("-713007610"), "1wJ4Qi", "KkU", bitmap_hash("3dQAC3pq"), "1970-01-01 08:00:00", "Z", ""), ("", "", bitmap_hash("oXom3"), percentile_hash("2110045409"), "V", "iO", bitmap_hash("yt3N4Uj"), "1970-01-01 08:00:00", "mI", "9"), ("", "BC", bitmap_hash(""), percentile_hash("-972806661"), "9", "P5Z5aLu", bitmap_hash("wnQj"), "1970-01-01 08:00:00", "", "");
INSERT INTO t2 VALUES ("", "I", bitmap_hash("TC"), percentile_hash("-1137877530"), "XHC", "C", bitmap_hash("Lu"), "1970-01-01 08:00:00", "aR", ""), ("", "O", bitmap_hash("U"), percentile_hash("-1756358574"), "W", "a", bitmap_hash("U"), "1970-01-01 08:00:00", "", "");
INSERT INTO t3 VALUES (NULL, false, false, 650339947, 0.23013163617587495), (false, false, false, 1404961331, 0.5895953489613989);
TRUNCATE TABLE t3;
INSERT INTO t0 VALUES (2, "", 245667195631.335, 0.8994760945908856, 0.33187444374520636, percentile_hash("-1151442401"), NULL, "1970-01-01 08:00:00", 0.18407982299507275, "1970-01-01 08:00:00", "XvMEKy", NULL);
INSERT INTO t1 VALUES (false, "xFNl", 0.000, 1.9649, -1, 0.1015930, false, 6.54, 11, "", 563192591774.3098422, bitmap_hash("1x"), "1970-01-01 08:00:00", percentile_hash("-1629371371"), "1970-01-01 08:00:00"), (false, "W", 0.008, 8.1999, -1355256112, 6.7358948, true, NULL, 12, "", 568879454914.2996485, bitmap_hash("f"), "1970-01-01 08:00:00", percentile_hash("429915860"), "1970-01-01 08:00:00");
-- SELECT subt1.c_1_4, subt1.c_1_7, t3.c_3_0, subt1.c_1_3, t3.c_3_2, subt1.c_1_8 FROM  t3 RIGHT OUTER JOIN (SELECT t1.c_1_0, t1.c_1_1, t1.c_1_2, t1.c_1_3, t1.c_1_4, t1.c_1_5, t1.c_1_6, t1.c_1_7, t1.c_1_8, t1.c_1_9, t1.c_1_10, t1.c_1_12, t1.c_1_14 FROM t1) subt1 ON t3.c_3_4 = subt1.c_1_7 AND t3.c_3_4 != subt1.c_1_7 AND t3.c_3_3 < subt1.c_1_0 AND t3.c_3_3 != subt1.c_1_0 GROUP BY subt1.c_1_4, subt1.c_1_7, t3.c_3_0, subt1.c_1_3, t3.c_3_2, subt1.c_1_8;
-- cardinality: 9
-- SELECT subt1.c_1_4, subt1.c_1_7, t3.c_3_0, subt1.c_1_3, t3.c_3_2, subt1.c_1_8 FROM  t3 RIGHT OUTER JOIN (SELECT t1.c_1_0, t1.c_1_1, t1.c_1_2, t1.c_1_3, t1.c_1_4, t1.c_1_5, t1.c_1_6, t1.c_1_7, t1.c_1_8, t1.c_1_9, t1.c_1_10, t1.c_1_12, t1.c_1_14 FROM t1) subt1 ON t3.c_3_4 = subt1.c_1_7 AND t3.c_3_4 != subt1.c_1_7 AND t3.c_3_3 < subt1.c_1_0 AND t3.c_3_3 != subt1.c_1_0 WHERE true GROUP BY subt1.c_1_4, subt1.c_1_7, t3.c_3_0, subt1.c_1_3, t3.c_3_2, subt1.c_1_8 UNION SELECT subt1.c_1_4, subt1.c_1_7, t3.c_3_0, subt1.c_1_3, t3.c_3_2, subt1.c_1_8 FROM  t3 RIGHT OUTER JOIN (SELECT t1.c_1_0, t1.c_1_1, t1.c_1_2, t1.c_1_3, t1.c_1_4, t1.c_1_5, t1.c_1_6, t1.c_1_7, t1.c_1_8, t1.c_1_9, t1.c_1_10, t1.c_1_12, t1.c_1_14 FROM t1) subt1 ON t3.c_3_4 = subt1.c_1_7 AND t3.c_3_4 != subt1.c_1_7 AND t3.c_3_3 < subt1.c_1_0 AND t3.c_3_3 != subt1.c_1_0 WHERE (NOT (true)) GROUP BY subt1.c_1_4, subt1.c_1_7, t3.c_3_0, subt1.c_1_3, t3.c_3_2, subt1.c_1_8 UNION SELECT subt1.c_1_4, subt1.c_1_7, t3.c_3_0, subt1.c_1_3, t3.c_3_2, subt1.c_1_8 FROM  t3 RIGHT OUTER JOIN (SELECT t1.c_1_0, t1.c_1_1, t1.c_1_2, t1.c_1_3, t1.c_1_4, t1.c_1_5, t1.c_1_6, t1.c_1_7, t1.c_1_8, t1.c_1_9, t1.c_1_10, t1.c_1_12, t1.c_1_14 FROM t1) subt1 ON t3.c_3_4 = subt1.c_1_7 AND t3.c_3_4 != subt1.c_1_7 AND t3.c_3_3 < subt1.c_1_0 AND t3.c_3_3 != subt1.c_1_0 WHERE ((true) IS NULL) GROUP BY subt1.c_1_4, subt1.c_1_7, t3.c_3_0, subt1.c_1_3, t3.c_3_2, subt1.c_1_8;
-- cardinality: 8

2. DROP DATABASE IF EXISTS g0;
CREATE DATABASE g0;
USE g0;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE t0 (c_0_0 BOOLEAN NULL ,c_0_1 CHAR(1) NULL ,c_0_2 HLL HLL_UNION NOT NULL ,c_0_3 FLOAT REPLACE_IF_NOT_NULL NULL ,c_0_4 VARCHAR(11) MIN NULL ) AGGREGATE KEY (c_0_0,c_0_1) DISTRIBUTED BY HASH (c_0_1,c_0_0) properties("replication_num"="3") ;
CREATE TABLE IF NOT EXISTS t1 (c_1_0 INT NULL ,c_1_1 DATETIME NULL ,c_1_2 TINYINT NULL ,c_1_3 DECIMAL(10, 0)  NULL ,c_1_4 BOOLEAN NOT NULL ,c_1_5 DECIMAL(4, 1)  NOT NULL ,c_1_6 DATETIME NOT NULL ,c_1_7 BOOLEAN NOT NULL ,c_1_8 SMALLINT NOT NULL ,c_1_9 CHAR(11) NULL ,c_1_10 TINYINT NOT NULL ,c_1_11 INT NOT NULL ,c_1_12 DECIMAL(10, 5)  NOT NULL ,c_1_13 DATE NULL ,c_1_14 DECIMAL(1, 0)  NULL ,c_1_15 INT NOT NULL ,c_1_16 DOUBLE NULL ,c_1_17 VARCHAR(21) NULL ) UNIQUE KEY (c_1_0,c_1_1,c_1_2,c_1_3,c_1_4,c_1_5,c_1_6) DISTRIBUTED BY HASH (c_1_2) properties("replication_num"="3") ;
CREATE TABLE t2 (c_2_0 BOOLEAN NULL ,c_2_1 DECIMAL(37, 9)  NOT NULL ,c_2_2 DECIMAL(14, 11)  NULL ,c_2_3 DATETIME NULL ,c_2_4 DECIMAL(14, 3)  NOT NULL ) DUPLICATE KEY (c_2_0) DISTRIBUTED BY HASH (c_2_0) properties("replication_num"="3") ;
CREATE TABLE IF NOT EXISTS t3 (c_3_0 DECIMAL(31, 13)  NULL ,c_3_1 FLOAT MIN NOT NULL ) AGGREGATE KEY (c_3_0) DISTRIBUTED BY HASH (c_3_0) properties("replication_num"="3") ;
INSERT INTO t3 VALUES (1439.1997793852481, 0.3328432533075397), (93500.7264521874585, 0.3328432533075397);
INSERT INTO t0 VALUES (false, "", hll_hash("6kFvoAW"), 0.8301904976875297, ""), (true, "", hll_hash("N92NVQ"), 0.20240589453598268, "WTIiv");
INSERT INTO t0 VALUES (false, "", hll_hash("gEfY5fYl"), 0.24302970232104337, "cwH9"), (true, "", hll_hash("aO"), 0.8692809906635914, "");
INSERT INTO t3 VALUES (18368.0796325091522, 0.8692809906635914);
INSERT INTO t1 VALUES (7, "1970-01-01 08:00:00", 3, 5434030963, true, 20.0, "1970-01-01 08:00:00", false, 8, "", 6, 29, 0.96808, "1970-01-01", 0, 30, 0.20240589453598268, "XBT"), (0, "1970-01-01 08:00:00", 7, 7237920448, false, 26.1, "1970-01-01 08:00:00", true, 2, "", 6, 19, 0.52707, "1970-01-01", 5, 31, 0.24302970232104337, "q");
INSERT INTO t1 VALUES (14, "1970-01-01 08:00:00", 5, 4034542838, false, 23.5, "1970-01-01 08:00:00", false, 11, "W", 4, 14, 0.63809, "1970-01-01", 4, 11, 0.9368270083063595, "valmy6A9");
INSERT INTO t3 VALUES (62066.4354903389503, 0.7747309626287371);
INSERT INTO t3 VALUES (12546.0883911294964, 0.24302970232104337), (2540.7719683675809, 0.7747309626287371), (46313.1741280791954, 0.0177184456986611);
INSERT INTO t3 VALUES (4305.1626840074364, 0.20240589453598268), (93321.9574114569649, 0.004329723411759301);
INSERT INTO t0 VALUES (NULL, "", hll_hash("OWF"), 0.9368270083063595, "bJFb4"), (true, "", hll_hash("G"), 0.7711559394514055, "EZpP3");
INSERT INTO t2 VALUES (true, 5566797919995940980.841808709, 6.09E-9, "1970-01-01 08:00:00", 96379981.786), (false, 369692023117948779.242780688, 8.2E-10, "1970-01-01 08:00:00", 71924353.764), (true, 6443440594408451239.649693320, 1.78E-9, "1970-01-01 08:00:00", 26577181.795);
INSERT INTO t3 VALUES (70515.6411491871246, 0.5330513326307974);
INSERT INTO t3 VALUES (22184.9739412524760, 0.9368270083063595), (62875.4628395843273, 0.17445222955966444);
INSERT INTO t3 VALUES (99614.9819391884971, 0.5944063829731546);
INSERT INTO t1 VALUES (NULL, "1970-01-01 08:00:00", 7, 1313381123, true, 64.4, "1970-01-01 08:00:00", true, 8, "L", 0, 26, 0.86407, "1970-01-01", 1, 1, 0.004329723411759301, "23kvKSmAe");
INSERT INTO t2 VALUES (true, 9485065314960854793.158162578, 6.02E-9, "1970-01-01 08:00:00", 88022627.398);
INSERT INTO t2 VALUES (false, 1334454280333171335.426456494, 6.60E-9, "1970-01-01 08:00:00", 49327414.950), (false, 8971834386822242152.215167720, NULL, "1970-01-01 08:00:00", 23192015.667), (false, 4208633664109673600.650651792, 3.21E-9, "1970-01-01 08:00:00", 34062180.935);
-- SELECT t3.c_3_0, t1.c_1_12, t1.c_1_16, t1.c_1_7, t1.c_1_9, t1.c_1_2, t2.c_2_4, t2.c_2_2, t1.c_1_8, t1.c_1_15, t3.c_3_1, t2.c_2_3, t2.c_2_1, t1.c_1_13, t1.c_1_5, t1.c_1_0, t1.c_1_3, t1.c_1_14, t1.c_1_11, t1.c_1_10, t2.c_2_0, t1.c_1_4 FROM t2, t3, t1 GROUP BY t3.c_3_0, t1.c_1_12, t1.c_1_16, t1.c_1_7, t1.c_1_9, t1.c_1_2, t2.c_2_4, t2.c_2_2, t1.c_1_8, t1.c_1_15, t3.c_3_1, t2.c_2_3, t2.c_2_1, t1.c_1_13, t1.c_1_5, t1.c_1_0, t1.c_1_3, t1.c_1_14, t1.c_1_11, t1.c_1_10, t2.c_2_0, t1.c_1_4;
-- cardinality: 364
-- SELECT t3.c_3_0, t1.c_1_12, t1.c_1_16, t1.c_1_7, t1.c_1_9, t1.c_1_2, t2.c_2_4, t2.c_2_2, t1.c_1_8, t1.c_1_15, t3.c_3_1, t2.c_2_3, t2.c_2_1, t1.c_1_13, t1.c_1_5, t1.c_1_0, t1.c_1_3, t1.c_1_14, t1.c_1_11, t1.c_1_10, t2.c_2_0, t1.c_1_4 FROM t2, t3, t1 WHERE true GROUP BY t3.c_3_0, t1.c_1_12, t1.c_1_16, t1.c_1_7, t1.c_1_9, t1.c_1_2, t2.c_2_4, t2.c_2_2, t1.c_1_8, t1.c_1_15, t3.c_3_1, t2.c_2_3, t2.c_2_1, t1.c_1_13, t1.c_1_5, t1.c_1_0, t1.c_1_3, t1.c_1_14, t1.c_1_11, t1.c_1_10, t2.c_2_0, t1.c_1_4 UNION SELECT t3.c_3_0, t1.c_1_12, t1.c_1_16, t1.c_1_7, t1.c_1_9, t1.c_1_2, t2.c_2_4, t2.c_2_2, t1.c_1_8, t1.c_1_15, t3.c_3_1, t2.c_2_3, t2.c_2_1, t1.c_1_13, t1.c_1_5, t1.c_1_0, t1.c_1_3, t1.c_1_14, t1.c_1_11, t1.c_1_10, t2.c_2_0, t1.c_1_4 FROM t2, t3, t1 WHERE (NOT (true)) GROUP BY t3.c_3_0, t1.c_1_12, t1.c_1_16, t1.c_1_7, t1.c_1_9, t1.c_1_2, t2.c_2_4, t2.c_2_2, t1.c_1_8, t1.c_1_15, t3.c_3_1, t2.c_2_3, t2.c_2_1, t1.c_1_13, t1.c_1_5, t1.c_1_0, t1.c_1_3, t1.c_1_14, t1.c_1_11, t1.c_1_10, t2.c_2_0, t1.c_1_4 UNION SELECT t3.c_3_0, t1.c_1_12, t1.c_1_16, t1.c_1_7, t1.c_1_9, t1.c_1_2, t2.c_2_4, t2.c_2_2, t1.c_1_8, t1.c_1_15, t3.c_3_1, t2.c_2_3, t2.c_2_1, t1.c_1_13, t1.c_1_5, t1.c_1_0, t1.c_1_3, t1.c_1_14, t1.c_1_11, t1.c_1_10, t2.c_2_0, t1.c_1_4 FROM t2, t3, t1 WHERE ((true) IS NULL) GROUP BY t3.c_3_0, t1.c_1_12, t1.c_1_16, t1.c_1_7, t1.c_1_9, t1.c_1_2, t2.c_2_4, t2.c_2_2, t1.c_1_8, t1.c_1_15, t3.c_3_1, t2.c_2_3, t2.c_2_1, t1.c_1_13, t1.c_1_5, t1.c_1_0, t1.c_1_3, t1.c_1_14, t1.c_1_11, t1.c_1_10, t2.c_2_0, t1.c_1_4;
-- cardinality: 348

3. DROP DATABASE IF EXISTS d0;
CREATE DATABASE d0;
USE d0;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE IF NOT EXISTS t0 (c_0_0 BOOLEAN NOT NULL ,c_0_1 BIGINT NOT NULL ,c_0_2 VARCHAR(1) NOT NULL ,c_0_3 VARCHAR(31) NOT NULL ,c_0_4 VARCHAR(11) NOT NULL ,c_0_5 DATETIME NOT NULL ,c_0_6 BIGINT NULL ,c_0_7 DATETIME NOT NULL ,c_0_8 LARGEINT NULL ,c_0_9 DECIMAL(1, 0)  NULL ,c_0_10 VARCHAR(1) NOT NULL ,c_0_11 CHAR(1) NULL ,c_0_12 BOOLEAN NOT NULL ,c_0_13 DATE NOT NULL ,c_0_14 BOOLEAN NULL ,c_0_15 BOOLEAN NOT NULL ,c_0_16 BOOLEAN NULL ,c_0_17 DATETIME NOT NULL ,c_0_18 DECIMAL(11, 5)  NULL ) UNIQUE KEY (c_0_0,c_0_1) DISTRIBUTED BY HASH (c_0_0,c_0_1) properties("replication_num"="3") ;
CREATE TABLE IF NOT EXISTS t1 (c_1_0 BIGINT NOT NULL ,c_1_1 BOOLEAN NOT NULL ,c_1_2 INT NOT NULL ,c_1_3 BOOLEAN NOT NULL ,c_1_4 CHAR(21) NOT NULL ,c_1_5 BOOLEAN NULL ,c_1_6 DATE NULL ) UNIQUE KEY (c_1_0,c_1_1) DISTRIBUTED BY HASH (c_1_0,c_1_1) properties("replication_num"="3") ;
INSERT INTO t0 VALUES (true, -1714531646, "", "aE", "G", "1970-01-01 08:00:00", 1301068946, "1970-01-01 08:00:00", -1599514599, 8, "", "", false, "1970-01-01", NULL, true, false, "1970-01-01 08:00:00", 6.67185);
INSERT INTO t1 VALUES (1301068946, true, 17, false, "", true, "1970-01-01"), (1301068946, true, 29, false, "Gi3redk6U", false, "1970-01-01");
INSERT INTO t0 VALUES (true, 1301068946, "", "", "6", "1970-01-01 08:00:00", -1599514599, "1970-01-01 08:00:00", 1301068946, NULL, "", "", false, "1970-01-01", false, false, true, "1970-01-01 08:00:00", 1.19230);
INSERT INTO t1 VALUES (-58278223, true, 23, true, "h", false, "1970-01-01");
INSERT INTO t1 VALUES (-1333363229, true, 2, false, "", true, "1970-01-01"), (-58278223, false, 5, true, "V9p2G7CIn", true, "1970-01-01");
INSERT INTO t1 VALUES (-58278223, true, 16, true, "B", true, "1970-01-01");
INSERT INTO t1 VALUES (1301068946, false, 20, true, "KIaNYkty2", true, "1970-01-01");
INSERT INTO t0 VALUES (false, -1085956619, "", "1tZ9f", "ol", "1970-01-01 08:00:00", 89113025, "1970-01-01 08:00:00", -1807854478, 4, "", "", false, "1970-01-01", false, true, true, "1970-01-01 08:00:00", 2.17970), (true, 1594737271, "", "N", "GASzDU", "1970-01-01 08:00:00", -563132038, "1970-01-01 08:00:00", 786585923, 6, "", "", false, "1970-01-01", true, false, NULL, "1970-01-01 08:00:00", 0.20536);
INSERT INTO t1 VALUES (-1923289504, true, 4, true, "", false, "1970-01-01"), (1368948252, true, 12, false, "", true, "1970-01-01");
INSERT INTO t0 VALUES (false, 568309835, "", "e", "", "1970-01-01 08:00:00", NULL, "1970-01-01 08:00:00", -563132038, 8, "", "", false, "1970-01-01", NULL, false, true, "1970-01-01 08:00:00", 1.61756), (true, -757429441, "", "", "4fbUI", "1970-01-01 08:00:00", -757429441, "1970-01-01 08:00:00", 1203291346, NULL, "", "", true, "1970-01-01", false, true, false, "1970-01-01 08:00:00", 2.03105);
INSERT INTO t1 VALUES (-330219809, false, 22, true, "", false, "1970-01-01");
INSERT INTO t0 VALUES (false, 739417030, "", "e", "5", "1970-01-01 08:00:00", 1368948252, "1970-01-01 08:00:00", -1051945497, 2, "", "", false, "1970-01-01", false, false, false, "1970-01-01 08:00:00", 2.23582), (true, -2075948076, "", "Qhem", "1j", "1970-01-01 08:00:00", NULL, "1970-01-01 08:00:00", 786585923, 2, "", "", true, "1970-01-01", false, false, true, "1970-01-01 08:00:00", 5.03708);
INSERT INTO t1 VALUES (-1327048348, false, 2, false, "33D", true, "1970-01-01"), (-1085956619, true, 3, true, "K5ExU", true, "1970-01-01");
INSERT INTO t1 VALUES (1303380246, false, 7, false, "P", NULL, NULL), (-1085956619, true, 11, false, "tLFY9pVJ", false, "1970-01-01");
INSERT INTO t1 VALUES (30004628, false, 6, true, "CQ3wi", true, "1970-01-01"), (-1765937200, true, 29, false, "4uw", false, "1970-01-01"), (-1085956619, true, 17, true, "Xg2r", false, "1970-01-01");
INSERT INTO t1 VALUES (1301068946, false, 0, true, "", true, "1970-01-01"), (-2016991838, false, 2, false, "gi", true, "1970-01-01");
INSERT INTO t1 VALUES (1546864557, true, 27, true, "TAbz", true, "1970-01-01"), (-1821178129, false, 5, false, "", false, "1970-01-01");
-- SELECT DISTINCT t1.c_1_0 FROM t1, t0;
-- cardinality: 14
-- SELECT DISTINCT * FROM (SELECT DISTINCT t1.c_1_0 FROM t1, t0 WHERE true UNION ALL SELECT DISTINCT t1.c_1_0 FROM t1, t0 WHERE (NOT (true)) UNION ALL SELECT DISTINCT t1.c_1_0 FROM t1, t0 WHERE ((true) IS NULL)) t;
-- cardinality: 13

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - main ASAN
 - mysql> select current_version();
+------------------------------+
| current_version()            |
+------------------------------+
| SQLANCER_01_ASAN 5aac710 |
+------------------------------+

**Comment by satanson** [comment]
EmptySetNode exists, some fragment run in pipeline engine, some in non-pipeline
```
| PLAN FRAGMENT 4                                                                                                             |
|  OUTPUT EXPRS:                                                                                                              |
|   PARTITION: UNPARTITIONED                                                                                                  |
|                                                                                                                             |
|   STREAM DATA SINK                                                                                                          |
|     EXCHANGE ID: 13                                                                                                         |
|     HASH_PARTITIONED: <slot 30>, <slot 33>, <slot 21>, <slot 29>, <slot 23>, <slot 34>                                      |
|                                                                                                                             |
|   12:AGGREGATE (update finalize)                                                                                            |
|   |  group by: 30: c_1_4, 33: c_1_7, 21: c_3_0, 29: c_1_3, 23: c_3_2, 34: c_1_8                                             |
|   |                                                                                                                         |
|   11:EMPTYSET 
```

**Comment by ZiheLiu** [comment]
Fixed by #3776.
